src/js/components/TableCell/TableCell.js

```diff
@@ -48,13 +48,39 @@ const TableCell = forwardRef(
     const tableContext = useContext(TableContext);
     const cellRef = useForwardedRef(ref);
     const containerRef = useRef();
+    const widthRef = useRef();
 
     useLayoutEffect(() => {
+      let resizeObserver;
+      const element = cellRef.current;
       if (onWidth) {
-        const { width } = cellRef.current.getBoundingClientRect();
-        onWidth(width);
+        if (typeof window !== 'undefined' && window.ResizeObserver) {
+          resizeObserver = new window.ResizeObserver((entries) => {
+            const entry = entries[0];
+            const { width } = entry.contentRect;
+            if (widthRef.current !== width) {
+              widthRef.current = width;
+              onWidth(width);
+            }
+          });
+          if (element) {
+            resizeObserver.observe(cellRef.current);
+          }
+        } else {
+          // fallback for server side rendering
+          const { width } = cellRef.current.getBoundingClientRect();
```

Comment:
Above we are getting width from `contentRect` which is getting the inner dimensions of the content (excludes borders and margins). Should we be doing the same here using `clientWidth`? Or is `getBoundingClientRect` what we want?

Reply:
That's a great question. getBoundingClientRect() is what it did before and does include borders. The ResizeObserverEntry.contentRect does not include borders as you noted and I should probably be using ResizeObserverEntry.borderBoxSize.  It doesn't matter for our existing styles EXCEPT for resizable columns. Resizable columns end up reporting a smaller width in contentRect (the difference is that `th` has padding.)